src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -844,9 +745,7 @@ protected virtual void MarkSecurityAttribute (SecurityAttribute sa, in Dependenc
 				return;
 			}
 
-			// Security attributes participate in inference logic without being marked.
-			Tracer.AddDirectDependency (sa, reason, marked: false);
```

Comment:
This will introduce a "hole" in the graph, right?
That is we will see that the type of the attribute is marked due to the security attribute, but there won't be any edge describing what marked the security attribute. I find this weird.